### PR TITLE
Switched to python connector config parser

### PR DESCRIPTION
**Pull request by sfc-gh-astus**
### Pre-review checklist
   * [x] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [x] I've added or updated automated unit tests to verify correctness of my new code.
   * [x] I've added or updated integration tests to verify correctness of my new code.
   * [ ] I've confirmed that my changes are working by executing CLI's commands manually.
   * [x] I've confirmed that my changes are up-to-date with the target branch.
   * [x] I've described my changes in the release notes.
   * [x] I've described my changes in the section below.

### Changes description
* Removed CliConfigManager class
* Switched to python connector default connection

### CODE REVIEW

**Review by sfc-gh-turbaszek — APPROVED** [review]

File: src/snowcli/config.py [hunk 1]
```diff
@@ -7,160 +7,131 @@
...
+        raise
+
+
+def _initialise_config(config_file: Path):
+    os.makedirs(os.path.dirname(config_file), exist_ok=True)
+    config_file.touch()
+    log.info(f"Created Snowflake configuration file at {CONFIG_MANAGER.file_path}")
+
+
+def _get_env_value(*path, key: str):
```
Comment (sfc-gh-mkeller) [inline src/snowcli/config.py:100]:
```suggestion
def _get_env_value(*path, key: str) -> str | None:
```

File: src/snowcli/config.py [hunk 2]
```diff
@@ -7,160 +7,131 @@
...
+        return env_variable
+    try:
+        return get_config_section(*path)[key]
+    except (KeyError, NonExistentKey, MissingConfigOptionError):
+        if default is not Empty:
+            return default
+        raise
+
+
+def _initialise_config(config_file: Path):
```
Comment (sfc-gh-mkeller) [inline src/snowcli/config.py:94]:
```suggestion
def _initialise_config(config_file: Path) -> None:
```

**Review by sfc-gh-mkeller — APPROVED** [review]
:ship: :ship:

File: src/snowcli/config.py (hunk 1 above)
Comment (sfc-gh-astus) [reply src/snowcli/config.py]: Added

File: src/snowcli/config.py (hunk 2 above)
Comment (sfc-gh-astus) [reply src/snowcli/config.py]: Added

**Pull request merged**